tests/coverage/unreachable/contradiction/expected

```diff
@@ -0,0 +1,6 @@
+coverage/unreachable/contradiction/main.rs, 5, FULL
+coverage/unreachable/contradiction/main.rs, 6, FULL
+coverage/unreachable/contradiction/main.rs, 7, PARTIAL
```

Comment:
The two checks correspond to the `if` (which is SUCCESS) and the condition (which is FAILURE). Since we aggregate checks by line and not by column, the line is being reported as PARTIAL.

Reply:
This case is confusing due to compiler optimizations: the entire `if x < 2 { ... }` statement can be optimized away because the compiler knows the condition will always be `false`. The MIR code obtained using the command shown [here](https://model-checking.github.io/kani/cheat-sheets.html#debug) doesn't show the optimized version, but the Rust Playground can if we replace the `kani::any()` call with a concrete value. This is what I got when replacing the call with `20`:
```
bb0: {
        _1 = const 20_u8;                // scope 0 at [src/main.rs:2:17: 2:19](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=330a3113cd82d916e617018557be49fc#)
        _2 = Gt(_1, const 5_u8);         // scope 1 at [src/main.rs:3:8: 3:13](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=330a3113cd82d916e617018557be49fc#)
        switchInt(move _2) -> [0: bb2, otherwise: bb1]; // scope 1 at [src/main.rs:3:8: 3:13](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=330a3113cd82d916e617018557be49fc#)
    }

    bb1: {
        goto -> bb4;                     // scope 1 at [src/main.rs:4:12: 4:17](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=330a3113cd82d916e617018557be49fc#)
    }

    bb2: {
        _4 = Lt(_1, const 10_u8);        // scope 1 at [src/main.rs:8:17: 8:23](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=330a3113cd82d916e617018557be49fc#)
        _3 = Not(move _4);               // scope 1 at [src/main.rs:8:9: 8:24](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=330a3113cd82d916e617018557be49fc#)
        switchInt(move _3) -> [0: bb4, otherwise: bb3]; // scope 1 at [src/main.rs:8:9: 8:24](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=330a3113cd82d916e617018557be49fc#)
    }

    bb3: {
        _5 = core::panicking::panic(const "assertion failed: x < 10"); // scope 1 at [src/main.rs:8:9: 8:24](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=330a3113cd82d916e617018557be49fc#)
                                         // mir::Constant
                                         // + span: [src/main.rs:8:9: 8:24](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=330a3113cd82d916e617018557be49fc#)
                                         // + literal: Const { ty: fn(&'static str) -> ! {core::panicking::panic}, val: Value(<ZST>) }
                                         // mir::Constant
                                         // + span: no-location
                                         // + literal: Const { ty: &str, val: Value(Slice(..)) }
    }

    bb4: {
        return;                          // scope 0 at [src/main.rs:10:2: 10:2](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=330a3113cd82d916e617018557be49fc#)
    }
```

Note that `bb1` has been replaced with an unconditional `goto` statement. The `goto` statement itself is reachable and reported as such. But the checks that come after it in the same basic block won't be. Because of that, we report `PARTIAL` instead of `FULL` (which is what most people would expect). This can be considered a spurious "failure" for coverage: not desirable but still sound.